evcouplings/compare/protocol.py

```diff
@@ -111,29 +139,230 @@ def print_pdb_structure_info(sifts_result, format_string=SIFTS_TABLE_FORMAT_STR,
+
+    # transform with the scaler
+    X = scaler.transform(X)
+
+    ecs.loc[:,output_column]=model.predict_proba(X)[:,1]
+    return ecs
+
+def complex_probability(ecs, scoring_model, use_all_ecs=False,
+                        score="cn", N_effL=None):
+    """
+    Adds confidence measure for complex evolutionary couplings
 
     Parameters
     ----------
-    **kwargs
-        See check_required in code below
+    ecs : pandas.DataFrame
+        Table with evolutionary couplings
+    scoring_model : {"skewnormal", "normal", "evcomplex"}
+        Use this scoring model to assign EC confidence measure
+    use_all_ecs : bool, optional (default: False)
+        If true, fits the scoring model to all ECs;
+        if false, fits the model to only the inter ECs.
+    score : str, optional (default: "cn")
+        Use this score column for confidence assignment
 
     Returns
     -------
-    SIFTSResult
-        Identified structures and residue index mappings
+    ecs : pandas.DataFrame
+        EC table with additional column "probability"
+        containing confidence measure
+    """
+    from evcouplings.couplings.pairs import add_mixture_probability
+
+    if use_all_ecs:
+        ecs = add_mixture_probability(
+            ecs, model=scoring_model
+        )
+    else:
+        inter_ecs = ecs.query("segment_i != segment_j")
+        intra_ecs = ecs.query("segment_i == segment_j")
+
+        intra_ecs = add_mixture_probability(
+            intra_ecs, model=scoring_model, score=score, N_effL=N_effL
+        )
+
+        inter_ecs = add_mixture_probability(
+            inter_ecs, model=scoring_model, score=score, N_effL=N_effL
+        )
+
+        ecs = pd.concat(
+            [intra_ecs, inter_ecs]
+        ).sort_values(
+            score, ascending=False
+        )
+
+    return ecs
+
+
+def _filter_structures(sifts_map, pdb_ids=None, max_num_hits=None, max_num_structures=None):
+
     """
+    Filters input SIFTSResult for specific pdb ids and/or number of hits
 
+    Parameters
+    ----------
+    sifts_map: SIFTSResult
+        Identified structures and residue index mappings
+    pdb_ids: list of str, optional (default: None)
+        List of PDB ids to be used for comparison
+    max_num_hits: int, optional (default: None)
+        Number of PDB hits to be used for comparison.
+        Different chains from the same PDB count as multiple hits.
+    max_num_structures: int, optional (default: None)
+        Number of unique PDB ids to be used for comparison.
+        Different chains from the same PDB count as one hit.
+
+    Returns
+    -------
+    SIFTSResult
+        Filtered identified structures and residue index mappings
+    """
     def _filter_by_id(x, id_list):
         x = deepcopy(x)
         x.hits = x.hits.loc[
             x.hits.pdb_id.isin(id_list)
         ]
         return x
 
+
+    # filter ID list down to manually selected PDB entries
+    if pdb_ids is not None:
+        pdb_ids = pdb_ids
```

Comment:
replace gone wrong? i think this can just go

Reply:
Yeah I'm not sure what this is doing here either